iroh-relay/src/server/http_server.rs

```diff
@@ -869,7 +876,7 @@ mod tests {
         let tls_config = make_tls_config();
 
         // start server
-        let mut server = ServerBuilder::new("127.0.0.1:0".parse().unwrap())
+        let mut server = ServerBuilder::new("127.0.0.1:0".parse().unwrap(), Default::default())
```

Comment:
might be nicer to make the metrics be a call 

eg
```rust
// default
ServerBuilder::new("127.0.0.1:0".parse().unwrap())
  .spawn();
// explicit
ServerBuilder::new("127.0.0.1:0".parse().unwrap())
  .metrics(metrics)
  .spawn();
```

Reply:
This would mean making it optional though, no? Currently metrics are required, which is why I added this on the constructor.
I can make them optional, it'll be `if let Some(metrics) = self.metrics { metrics.foo.inc(); }` then at each place where metrics are used.

Reply:
I didn't mean to make them optional, just construct them as `default` if none are passed in

Reply:
Changed that to use the builder method instead of Default in the constructor.